### DISCUSSION TITLE
Get authorization code param from oauth provider callback

### DISCUSSION THREAD

**Discussion by jilherme** [nextauthjs/next-auth#8499] [Help]
In our app we have separated front/backend. I'm using NextJs and trying to use next-auth with cognito. 
My backend is using cognito to connect with google, facebook, etc., and say he needs the ?code that is returned in the callback url to the frontend after successful login like `http://localhost:3000/api/auth/callback/cognito?code=xxxxxxxx`.

I have followed several stackeoverflow, issues and discussions, but I couldn't find an proper way to do it. My question is: there is a way to do it? Should we be doing in another way?

**Comment by arbKow** [comment]
I have same issue! I'm looking for answer but nothing...
I ened up on writing custom provider for google instead of using the one that next-auth already have. However i'm not sure if that works

**Reply by jilherme** [reply]
I think for our need it's better to use iron-session, it provides more low level handling. The problem for me it's that they didn't ship full support for `/app` router :(
